docs/content/guides/dagster/managing-ml.mdx

```diff
@@ -0,0 +1,271 @@
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=basic_schedule_start endbefore=basic_schedule_end
+from dagster import AssetSelection, define_asset_job, ScheduleDefinition
+
+ml_asset_job = define_asset_job("ml_asset_job", AssetSelection.groups("ml_asset_group"))
+
+basic_schedule = ScheduleDefinition(job=ml_asset_job, cron_schedule="0 9 * * *")
+```
+
+### Monitoring
+
+Integrating your machine learning models into Dagster allows you to see when the model and its data dependencies were refreshed, or when a refresh process has failed. By using Dagster to monitor performance changes and process failures on your ML model, it becomes possible to set up remediation paths, such as automated model retraining, that can help resolve issues like model drift.
+
+In this example, the model is being evaluated against the previous model’s accuracy. If the model’s accuracy has improved, the model is returned for use in downstream steps, such as inference or deploying to production.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=conditional_monitoring_start  endbefore=conditional_monitoring_end
+from dagster import RunRequest, asset, Output
+from sklearn import linear_model
+
+
+@asset
+def other_ml_model(training_data, test_data, model_accuracy):
+    reg = linear_model.LinearRegression()
+    reg.fit(training_data)
+    new_model_accuracy = reg.score(test_data)
+    if new_model_accuracy > model_accuracy:
+        yield Output(reg, metadata={"model_accuracy": new_model_accuracy})
+```
+
+We can also monitor the score or other information of a model through alerts, such as email or Slack. In the example below, we set up a Slack message to post the score of `ml_model` upon success. The message can be customized to include information such as which features were most influential in model training, the volume of data that the model was trained on, or in this case, the score of the model.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=success_slack_start  endbefore=success_slack_end
+from dagster import asset_sensor, RunRequest, asset
+from sklearn import linear_model
+
+
+@asset
+def ml_model(training_data, test_data, model_accuracy):
+    reg = linear_model.LinearRegression()
+    reg.fit(training_data)
+    new_model_accuracy = reg.score(test_data)
+    if new_model_accuracy > model_accuracy:
+        yield Output(reg, metadata={"model_accuracy": new_model_accuracy})
+```
+
+We can also set up a sensor that triggers if an asset fails to materialize. Alerts can be customized and sent through e-mail or natively through Slack. In this example, a Slack message is sent anytime the `ml_job` fails.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=fail_slack_start   endbefore=fail_slack_end
+import os
+from dagster import make_slack_on_run_failure_sensor, define_asset_job
+
+ml_job = define_asset_job("ml_training_job", selection=[ml_model])
```

Comment:
This is a little confusing, because earlier we recommended auto-materialize policies,  but here we're expecting they'll run it as a scheduled job.  This is definitely a gap in the product.  I can't think of an obvious answer though for what we should do instead. Maybe we can chat about it?

Reply:
sounds good